spikeinterface/sortingcomponents/tests/test_dimensionality_reduction.py

```diff
@@ -0,0 +1,78 @@
+
+import pytest
+
+import numpy as np
+
+import spikeinterface as si
+import spikeinterface.extractors as se
+from spikeinterface.sortingcomponents.dimensionality_reduction import TemporalPCA
+from spikeinterface.sortingcomponents.peak_pipeline import run_peak_pipeline
+from spikeinterface.sortingcomponents.peak_detection import detect_peaks
+
+
+def test_dimensionality_reduction(tmp_path):
+    local_path = si.download_dataset(remote_path='mearec/mearec_test_10s.h5')
+    recording, sorting = se.read_mearec(local_path)
+
+    local_radius_um = 100
+
+    model_path = tmp_path / "buffer_pca.pkl"
+    temporal_pca = TemporalPCA(recording, model_path=model_path, local_radius_um=local_radius_um)
+
+    n_components = 3
+    job_kwargs = dict(n_jobs=1, chunk_size=10000, progress_bar=True)
+    n_peaks = recording.get_num_channels() * 1e3 # Heuristic for extracting around 1k waveforms per channel
+    peak_selection_params = dict(method="uniform", select_per_channel=True,  n_peaks=n_peaks)
+    detect_peaks_params = dict(method='by_channel', peak_sign='neg', detect_threshold=5, exclude_sweep_ms=0.1)
+    temporal_pca.fit(recording, n_components, detect_peaks_params, peak_selection_params, job_kwargs)
+
+    steps = [temporal_pca]
+    
+    peaks, projected_waveforms = detect_peaks(recording, pipeline_steps=steps)
```

Comment:
Can you add 2 options?

1. fitting with `n_jobs=2`
2. running `detect_peaks` in parallel (`n_jobs=2`)

This makes sure that the serialization is working

Reply:
I added now a new extra test case that does this two things.